wit/ephemeral/wasi-mongo.wit

```diff
@@ -0,0 +1,42 @@
+
+use * from types
+
+type document = list<u8> 
+
+// Create operations
+
+// Insert a document into a collection 
+insert-one: function(db: string, collection: string, doc: document) -> expected<_, error>
+
+// Insert many documents into a collection
+insert-many: function(db: string, collection: string, docs: list<document>) -> expected<_, error> 
+
+// Read operations
+
+// Get a list of documents from the collection
+// TODO (nitishm): Add support for query params
+find: function(db: string, collection: string, filter: document) -> expected<list<document>, error>
+
+// Update operations
+
+// Update one document in the collection
+update-one: function(db: string, collection: string, filter: document, doc: document) -> expected<_, error>
```

Comment:
How about instead of `-one` and `-many` functions we would only have one for each CRUD operation that accepted a `list<collection>` ?

Reply:
I thought about that but I was thinking about how close we want to keep the exported APi to the actual Mongo CRUD API since mongo has separate APIs for XXX-one, XXX-many? What are your thoughts?

Reply:
I definitely understand the reason to keep the API close to the Mongo API.
There are two things I'd consider:

- ideally, we'd like this to be a "document database" interface, rather than a "Mongo" interface
- you can think at the interface as to a an FFI layer (foreign function interface). Where possible you want to have as few functions as possible that an implementor has to write.

Both of those would make me lean towards a more generic document API, rather than one specialized for Mongo, with as few functions as possible.

What do you think?

Reply:
That makes sense. I wasnt aware that we were thinking beyond mongo( and mongo like) backends. I will shrink it down in that case :+1: